content/client-lib-development-guide/features.textile

```diff
@@ -508,6 +519,268 @@ h3(#eventemitter). EventEmitter mixin / interface
 * @(RTE5)@ @EventEmitter#off@ deregisters a listener. If called with a specific event and a listener, it removes all registrations that match both the given listener and the given event; if called only with a listener, it removes all registrations matching the given listener, regardless of whether they are associated with an event or not; if called with no arguments, it removes all registrations, for all events and listeners
 * @(RTE6)@ @EventEmitter#emit@ emits an event, calling registered listeners with the given event name and any other given arguments. If an exception is raised in any of the listeners, the exception is caught by the @EventEmitter@ and the exception is logged to the Ably logger. Tests must exist to ensure exceptions raised in client code do not propagate and inhibit other event processing within the client library
 
+h2(#state-conditions-and-operations). State conditions and operations
+
+@Connection@ operation effects depending on current @Connnection.state@:
+
+<table id="connection-states-operations">
+
+<tr>
+  <th></th>
+  <th>Initialized</th>
+  <th>Connecting</th>
+  <th>Connected</th>
+  <th>Disconnected</th>
+  <th>Suspended</th>
+  <th>Closing</th>
+  <th>Closed</th>
+  <th>Failed</th>
+</tr>
+
+<tr>
+  <td>@connect@</td>
+
+  <!-- When INITIALIZED -->
+  <td>"RTN11a":#RTN11a</td>
+  <!-- When CONNECTING -->
+  <td>No-op</td>
+  <!-- When CONNECTED -->
+  <td>No-op</td>
+  <!-- When DISCONNECTED -->
+  <td>No-op</td>
+  <!-- When SUSPENDED -->
+  <td>No-op</td>
+  <!-- When CLOSING -->
+  <td>"RTN11b":#RTN11b</td>
+  <!-- When CLOSED -->
+  <td>"RTN11a":#RTN11a</td>
+  <!-- When FAILED -->
+  <td>No-op</td>
```

Comment:
Correct; the `CLOSED` and `FAILED` states are "terminal" only on the sense that there are never autonomous transitions from those states. If there is an explicit `connect()` by the client, then a new connection will be attempted, even if from the `FAILED` state. The connection might have been forcibly closed because the account is blocked, for example, but the reconnection attempt can succeed one the block is removed.